pkg/pillar/cmd/volumemgr/updatestatus.go

```diff
@@ -268,9 +277,17 @@ func doUpdateContentTree(ctx *volumemgrContext, status *types.ContentTreeStatus)
 		// TODO: next stage, load disk images as well
 		if status.Format == zconfig.Format_CONTAINER {
 			// get all of the blobs
-			if err := containerd.LoadBlobs(lookupBlobStatuses(ctx, status.Blobs...), status.RelativeURL); err != nil {
-				log.Errorf("doUpdateContentTree(%s) unable to load blobs into containerd: %v", status.Key(), err)
-				status.SetErrorWithSource(fmt.Sprintf("unable to load blobs into containerd: %v", err),
+			loadedBlobs, err := ctx.casClient.IngestBlob(lookupBlobStatuses(ctx, status.Blobs...)...)
+			for _, loadedBlob := range loadedBlobs {
+				if loadedBlob.State == types.LOADED && loadedBlob.HasVerifierRef {
+					MaybeRemoveVerifyImageConfig(ctx, loadedBlob.Sha256)
+					loadedBlob.HasVerifierRef = false
+				}
+				publishBlobStatus(ctx, loadedBlob)
+			}
```

Comment:
I see an issue here. You ingest the blobs - I had to walk through a comparison with the existing logic, but it appears to do the right thing, including adding the right labels - but you never create the image. As far as I can tell, it only happens with `CreateImage()`, which only gets called when creating a snapshot via `PrepareContainerRootDir()`.

This leaves the content blobs referenced by the manifest, manifest referenced by the index... and nothing referencing the image. You need to create the image immediately upon loading the blobs. That was why the original `LoadBlobs()` has an argument `status.RelativeURL` and creates the image reference.

Reply:
Refactored the GC logic of blobs. VolumeMgr is not responsible for GCing the blobs now. Instead, while handling a ContentTree we loaded the blobs and added a reference. And now while deleting the ContentTree, we'll remove that reference which will kick in ctrd's GC and remove all the unused blobs.

Reply:
Looks good.